### PR TITLE
Register CPU/CUDA fuser dynamically

### PR DESCRIPTION
**Pull request by zdevito**
This avoids a bunch of conditional compilation logic

### CODE REVIEW

**Comment by zou3519** [comment]
Taking a look. cc @mruberry too (I think you wrote most of the conditional compilation logic originally?)

File: tools/build_variables.py [hunk 1]
```diff
@@ -99,6 +99,14 @@
     "torch/csrc/jit/tracer.cpp",
     "torch/csrc/utils/tensor_flatten.cpp",
     "torch/csrc/utils/variadic.cpp",
+    "torch/csrc/jit/fuser/kernel_cache.cpp",
+    "torch/csrc/jit/fuser/compiler.cpp",
+    "torch/csrc/jit/fuser/executor.cpp",
+    "torch/csrc/jit/fuser/codegen.cpp",
+    "torch/csrc/jit/fuser/fallback.cpp",
+    "torch/csrc/jit/fuser/cpu/fused_kernel.cpp",
+    "torch/csrc/jit/fuser/cpu/dynamic_library_unix.cpp",
```
Comment (zou3519) [inline tools/build_variables.py]: Is this list for libtorch only? Do we support libtorch on windows? (we're including dynamic_library_unix here so it sounds like no but just checking)

File: tools/build_variables.py [hunk 2]
```diff
@@ -171,8 +176,6 @@ def torch_vars():
             "-Icaffe2/torch/csrc",
             "-Icaffe2/torch/csrc/nn",
             "-Icaffe2/torch/lib",
-            "-DUSE_CPU_FUSER_FBCODE=1",
```
Comment (zou3519) [inline tools/build_variables.py]: Are we still disabling the CPU Fuser in fbcode? I'm not sure where that logic lives... (hopefully it is in a buck file somewhere)

File: tools/build_variables.py (hunk 1 above)
Comment (zdevito) [reply tools/build_variables.py]: This is the buck build. No windows.

File: tools/build_variables.py (hunk 2 above)
Comment (zdevito) [reply tools/build_variables.py]: I copied the rules for what was there. According to this code it is enabled in fbcode.

**Review by zou3519 — APPROVED** [review]
lgtm, but the windows build is complaining about a line that doesn't exist anymore:

```
19:50:32 C:\Jenkins\workspace\pytorch-builds\pytorch-win-ws2016-cuda9-cudnn7-py3-build\torch/csrc/jit/fuser/cuda/resource_strings.h(55): fatal error C1020: unexpected #endif
```
which is really strange

**Comment by zdevito** [comment]
I know, I am not sure what is going on there.

**Comment by mruberry** [comment]
Thanks for cc, it's great to see the build logic generalized and simplified like this.

**Pull request closed without merging**